executor/index_lookup_hash_join.go

```diff
@@ -506,7 +506,7 @@ func (iw *indexHashJoinInnerWorker) getNewJoinResult(ctx context.Context) (*inde
 
 func (iw *indexHashJoinInnerWorker) buildHashTableForOuterResult(ctx context.Context, task *indexHashJoinTask, h hash.Hash64) {
 	buf, numChks := make([]byte, 1), task.outerResult.NumChunks()
-	task.lookupMap = newRowHashMap(task.outerResult.Len())
+	task.lookupMap = newUnsafeHashTable(task.outerResult.Len())
```

Comment:
@XuHuaiyu 
the `newUnsafeHashTable` has a changed behavior that it does not guarantee the order of entries from the same hash value. 
I thought it's ok for Index Hash Join but wish you to reconfirm that.

Reply:
The comment on `indexHashJoin` said that it does not promise the output order.